### ISSUE TITLE
Multiple Executes

### ISSUE THREAD

**Issue by paul-gauci** [natemcmaster/CommandLineUtils#254]
**Describe the bug**
I am trying to implement an infinite loop to process commands one after the other, but the 
`OnExecute` is only execute once.

**To Reproduce**
1. Using this version of the library '2.3.4'
```

    [Command(Description = "To Maintain Data")]
    public class DataCommand
    {
        
        public int OnExecute(CommandLineApplication app)
        {
            app.ShowHelp();
            Console.WriteLine("few");
            return 1;
        }
    }
```

```
    [Command("test", Description="", ThrowOnUnexpectedArgument = false), Subcommand(typeof(DataCommand))]
    public class Program
    {
        static int Main(string[] args)
        {

            var services = new ServiceCollection()
                .AddSingleton(PhysicalConsole.Singleton)
                .BuildServiceProvider();

            CommandLineApplication<Program> app = new CommandLineApplication<Program>();

            app.Conventions
                .UseDefaultConventions()
                .UseConstructorInjection(services);
            
            var exitProgram = false;
            while(!exitProgram)
            {
                string[] messageReceived;
                if(args.Length > 0)
                {
                    messageReceived = args;
                    exitProgram = true;
                }
                else
                {
                    messageReceived = Console.ReadLine().Split();
                }
                var exitCode = app.Execute(messageReceived);
                Console.WriteLine(exitCode);
            }

            return 0;
        }

        private int OnExecute(CommandLineApplication app)
        {
            app.ShowHelp();
            
            return 0;
        }
    }
```

**Expected behavior**
I am expecting that if i pass `Data` the first time it shows the 'help'. If I write it again, I am expecting to show the 'help' once again. From my understanding, the 'OnExecute' seems that have not been executed the second time round.

**Comment by paul-gauci** [comment]
For the time being, I am initialising a new CommandLineApplication<Program> instance, and it works, but I do not think it is a workaround

**Comment by paul-gauci** [comment]
Just saw issue #253, therefore duplicate issie

**Comment by natemcmaster** [comment]
FYI #253 will be fixed in 2.4. I'm releasing an RC soon to NuGet if you'd like to test it out.
